### ISSUE TITLE
OSError: [Errno 22] Invalid argument

### ISSUE THREAD

**Issue by arbitrager44** [pyinstaller/pyinstaller#4618]
This appears to be different than #3679 as the error results aren't quite the same. Error is thrown during freezing process. I have previously frozen another version of the same application if that matters. I also tried freezing a different application without trouble.

Complete output with error:
```
81 INFO: PyInstaller: 3.5
84 INFO: Python: 3.7.5
91 INFO: Platform: Windows-10-10.0.18362-SP0
94 INFO: wrote C:\Users\arbit\OneDrive\Documents\python\sweater\sm3000_txb.spec
96 INFO: UPX is not available.
99 INFO: Extending PYTHONPATH with paths
['C:\\Users\\arbit\\OneDrive\\Documents\\python\\sweater',
 'C:\\Users\\arbit\\OneDrive\\Documents\\python\\sweater']
99 INFO: checking Analysis
466 INFO: checking PYZ
471 INFO: Building because PYZ-00.toc is bad
471 INFO: Building PYZ (ZlibArchive) C:\Users\arbit\OneDrive\Documents\python\sweater\build\sm3000_txb\PYZ-00.pyz
5183 INFO: Building PYZ (ZlibArchive) C:\Users\arbit\OneDrive\Documents\python\sweater\build\sm3000_txb\PYZ-00.pyz completed successfully.
5291 INFO: checking PKG
5295 INFO: Building because PKG-00.toc is bad
5301 INFO: Building PKG (CArchive) PKG-00.pkg
>
>Traceback (most recent call last):
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\arbit\AppData\Local\Programs\Python\Python37-32\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\__main__.py", line 111, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\__main__.py", line 63, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\building\build_main.py", line 844, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\building\build_main.py", line 791, in build
    exec(code, spec_namespace)
  File "C:\Users\arbit\OneDrive\Documents\python\sweater\sm3000_txb.spec", line 29, in <module>
    console=True )
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\building\api.py", line 435, in __init__
    upx_exclude=self.upx_exclude
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\building\api.py", line 197, in __init__
    self.__postinit__()
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\building\datastruct.py", line 158, in __postinit__
    self.assemble()
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\building\api.py", line 284, in assemble
    pylib_name=pylib_name)
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\archive\writers.py", line 334, in __init__
    super(CArchiveWriter, self).__init__(archive_path, logical_toc)
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\archive\writers.py", line 61, in __init__
    self._add_from_table_of_contents(logical_toc)
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\archive\writers.py", line 86, in _add_from_table_of_contents
    self.add(toc_entry)  # The guts of the archive.
  File "c:\users\arbit\appdata\local\programs\python\python37-32\lib\site-packages\PyInstaller\archive\writers.py", line 416, in add
    buf = fh.read(16*1024)
OSError: [Errno 22] Invalid argument
```

**Comment by Legorooj** [comment]
@nithinreddyy can you take a look at this?

**Comment by nithinreddyy** [comment]
Hi @arbitrager44, will you please try to remove all unwanted files and try that again. It helps.

**Comment by bwoodsend** [comment]
@arbitrager44 Can we have your build command (e.g. `PyInstaller myscript.py`) so we can verify you're not doing something completely unorthodox?

I also notice your working directory `C:\Users\arbit\OneDrive\Documents\python\sweater` - is that `OneDrive` folder not a special reserved folder for syncing Documents with the cloud? 

And I'm fairly certain that #3679 has nothing to do with this - that's appears to be just a misunderstanding of how to use PyInstaller's command line interface.

@arbitrager44 You're going to have to clarify what you mean in your last comment. Do you mean source files or automatically generated files from PyInstaller?

**Comment by arbitrager44** [comment]
Hi. This problem went away on its own a few days later. I don't know what changed. I did clear out the `dist` folder as well as the `.spec` files, but I did it several times before it worked, so that was probably coincidental. Most likely this was just some internal conflict or corruption on my side.  

On the OneDrive bit, the only time I've had trouble with that directory is when OneDrive is actively trying to backup a file at the same time I'm trying write/re-write it, but then you get an access error, not this OS error. When I finally got the installer to run properly, it was from the same directory.

@bwoodsend, I only used `pyinstaller sm3000_txb.py` to launch.
